### PR TITLE
Initial draft of StaticWeb

### LINKED ISSUE #48: staticweb middleware
We need to implement staticweb middleware!

### PR DESCRIPTION
**Pull request by gholt**
Closes #48

### CODE REVIEW

**Comment by gholt** [comment]
Still in progress; do not merge. Need to write some tests at a minimum.

**Comment by gholt** [comment]
Good for reviewing now.

File: proxyserver/middleware/staticweb.go [hunk 1]
```diff
@@ -0,0 +1,386 @@
...
+	subreq, err := http.NewRequest("GET", fmt.Sprintf("/v1/%s/%s/%d%s", url.PathEscape(s.account), url.PathEscape(s.container), status, s.webError), nil)
+	if err != nil {
+		srv.StandardResponse(writer, status)
+		return
+	}
+	subreq.Header.Set("X-Auth-Token", request.Header.Get("X-Auth-Token"))
+	subrec := httptest.NewRecorder()
+	s.ctx.Subrequest(subrec, subreq, "staticweb", false)
+	subresp := subrec.Result()
+	subresp.Body.Close()
```
Comment (gholt) [inline proxyserver/middleware/staticweb.go]: I think this is wrong and I meant to defer it.
